<launch>
    <arg name="drone_id"/>
    <arg name="odom"/>

  <node pkg="nodelet" type="nodelet" name="manager" args="manager" output="screen">
    <param name="num_worker_threads" value="16"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="mapping" args="load mapping/Nodelet manager" output="screen">
    <!-- camera parameters -->
    <rosparam command="load" file="$(find mapping)/config/camera.yaml"/>
    <!-- remap topics -->
    <remap from="~odom" to="/drone_$(arg drone_id)_odom"/>
    <remap from="~depth" to="/drone_$(arg drone_id)_depth"/>
    <remap from="~gridmap" to="/drone_$(arg drone_id)_gridmap"/>
    <remap from="~gridmap_inflate" to="/drone_$(arg drone_id)_gridmap_inflate"/>
    <param name="down_sample_factor" value="2"/>
    <!-- depth filter parameters -->
    <param name="depth_filter_tolerance" value="0.15"/>
    <param name="depth_filter_mindist" value="0.2"/>
    <param name="depth_filter_margin" value="2"/>
    <!-- raycasting parameters -->
    <param name="p_min" value="-199"/>
    <param name="p_max" value=" 220"/>
    <param name="p_hit" value="  62"/>
    <param name="p_mis" value="  62"/>
    <param name="p_occ" value=" 139"/>
    <param name="p_def" value="-199"/>
    <!-- mapping parameters -->
    <param name="resolution" value="0.15"/>
    <param name="local_x" value="25"/>
    <param name="local_y" value="25"/>
    <param name="local_z" value="5"/>
    <param name="inflate_size" value="1"/> <!-- 1-->
    <param name="use_mask" value="false"/>

    <param name="use_global_map" value="false" />
    <param name="x_length" value="25.0"/> <!-- 25-->
    <param name="y_length" value="25.0"/>
    <param name="z_length" value="3.0"/>
    <remap from="~global_map" to="/global_map" />
  </node>
</launch>